### PR TITLE
Code Guidelines and MGC

### PR DESCRIPTION
**Pull request by tpsatish95**
- [x] Add pre-commit hooks to ensure pep8 and 
- [x] Add contributing guidelines
- [x] Restructure package 
- [x] Add rank_distance_matrix function and tests
- [x] Add pep8 badge
- [x] Add code coverage badge

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1 +1,42 @@
...
+  - A quick summary and/or background
+  - Steps to reproduce
+    - Be specific!
+    - Give sample code
+  - What you expected would happen
+  - What actually happens
+  - Notes (possibly including why you think this might be happening, or stuff you tried that didn't work, screenshots, etc.)
+
+## Proposing new features / fixes
+Pull requests are the best way to propose changes to the codebase (we use [GitHub Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
```
Comment (rguo123) [inline CONTRIBUTING.md]: do we have github flow setup?

Comment (tpsatish95) [reply CONTRIBUTING.md]: I think it just a guideline to follow and not something to setup.

Comment (rguo123) [reply CONTRIBUTING.md]: Oh i see

File: .coveralls.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
+repo_token: ZUTKvNcViYlweUGjtm22HmNsjN2CX5Pqf
```
Comment (rguo123) [inline .coveralls.yml]: Apparently public repos don't need repo_token. I think we can delete this file. 
https://github.com/lemurheavy/coveralls-public/issues/208

**Comment by rguo123** [comment]
@sampan501 i had an unresolved comment. Do you mind dismissing your review for now?

**Comment by sampan501** [comment]
> @sampan501 i had an unresolved comment. Do you mind dismissing your review for now?

Done

File: .coveralls.yml (hunk 2 above)
Comment (tpsatish95) [reply .coveralls.yml]: Oh alright! Will remove now.

Comment (tpsatish95) [reply .coveralls.yml]: Done.

**Pull request merged**